extension/src/popup/helpers/soroban.ts

```diff
@@ -168,22 +159,19 @@ export const getOpArgs = (fnName: string, args: SorobanClient.xdr.ScVal[]) => {
 const isSorobanOp = (operation: HorizonOperation) =>
   SOROBAN_OPERATION_TYPES.includes(operation.type);
 
-const getRootInvocationArgs = (hostFn: SorobanClient.xdr.HostFunction) => {
+const getTxArgs = (hostFn: SorobanClient.xdr.HostFunction) => {
   if (!hostFn) {
     return null;
   }
 
-  const txAuth = hostFn.auth();
+  const [_contractId, methodSymbol, ...args] = hostFn.args().value() as SorobanClient.xdr.ScVal[];
 
-  if (!txAuth.length) {
+  if (!args.length) {
     return null;
   }
 
-  const {
-    _attributes: attrs,
-  } = (txAuth[0].rootInvocation() as unknown) as RootInvocation;
-
-  const fnName = attrs.functionName.toString();
+  const fnName = (methodSymbol.value() as string).toString();
```

Comment:
this kinda sucks but the type isn't generic even though its from a union so I think you have to cast these until they become generic in the sdk.